packages/block-library/src/table-of-contents/hooks.js

```diff
@@ -105,10 +105,9 @@ function getLatestHeadings( select, clientId ) {
 				latestHeadings.push( {
 					// Convert line breaks to spaces, and get rid of HTML tags in the headings.
 					content: stripHTML(
-						headingAttributes.content.replace(
-							/(<br *\/?>)+/g,
-							' '
-						)
+						headingAttributes.content
+							.toString()
+							.replace( /(<br *\/?>)+/g, ' ' )
```

Comment:
We could actually use `toPlainText` as well now, then replace all `\n` with spaces. But right now we cannot absolutely guarantee that it will be a rich text value, I need to make a PR for that.

Reply:
I'm torn here because this code is already doing the very thing we _don't_ want to do: treat HTML as a string and perform risky parsing on it.

while it was the case before that the `content` attribute was a string, it was still an _HTML_ string, so we could have found a `RichText` way to extract the plaint text content.

in this specific case I'd rather we use `toPlainText()` and then ponder what to do about stringly-based code. for example, maybe adding the string methods only helps to perpetuate problems like this that are essentially longstanding type errors. having things crash early in the release cycle gives us a chance to update dependent code so that it remains more resilient in the future.

Reply:
The only reason why we'd add those methods would be to avoid errors

Reply:
more than simply call `.toString()` do we need to wrap all these in `RichTextData.from( headingAttributes.content )`?

for PRs like this we could then directly modify the content, and perform more semantic operations like `.removeTags( 'br' )`, or at a minimum call `.toString()` after jumping back into the `RichText` world